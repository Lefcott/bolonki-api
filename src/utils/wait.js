/**
 * Waits the specified milliseconds and resolves
 * @param {number} milliseconds Milliseconds to wait
 */
export const wait = async milliseconds => new Promise(resolve => setTimeout(resolve, milliseconds));
