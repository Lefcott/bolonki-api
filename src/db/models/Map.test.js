import 'jest';
import 'globals';
import 'babel-polyfill';
import * as util from 'util';
import rp from 'request-promise';
import { wait } from 'utils/wait';
import { exec, spawn } from 'child_process';
import projectDir from 'utils/projectDir';
import { gql } from 'apollo-server-express';
import { print } from 'graphql';

const pExec = util.promisify(exec);

const API = 'http://localhost:4000/graphql';
const CMD_SEED_DATABASE = `${projectDir}/scripts/seed.sh Maps`;
let api;

describe('Getting map items', () => {
  beforeAll(async () => {
    log('Spawning API');
    api = spawn('npm', ['start']);
    await wait(2000);
    await pExec(CMD_SEED_DATABASE);
  });
  afterAll(async () => {
    log('Killing API');
    await api.kill();
    log('API killed');
  });

  it('Retrieves the maps', async () => {
    const query = gql`
      query {
        maps {
          Name
          Polygons {
            Sides
            AdjacentPolygons
          }
        }
      }
    `;

    const response = await rp({ method: 'POST', uri: API, body: { query: print(query) }, json: true });
    expect(response).toMatchSnapshot();
  });
});
