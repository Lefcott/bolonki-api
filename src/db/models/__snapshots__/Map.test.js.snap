// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Getting map items Retrieves the maps 1`] = `
Object {
  "data": Object {
    "maps": Array [],
  },
}
`;
